transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,117 @@
+* The administrator needs to receive notifications when the Transactions subsystem is causing delays in the graceful shutdown of WildFly, achieved through integration with the existing logging mechanism. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to the administrator in an understandable manner without exposing internal technical implementation details
+
+=== Nice-to-Have Requirements
+
+* Introducing an asynchronous graceful shutdown would enable all ServerActivity implementations to receive a pre-suspend/suspend signal concurrently. In this way, the timeout will have the same duration for all ServerActivity implementations.
+* [The sequence to suspend](https://github.com/wildfly/wildfly-core/blob/22.0.1.Final/server/src/main/java/org/jboss/as/server/suspend/SuspendController.java#L62) ServerActivity implementations during WildFly’s graceful shutdown should be Last In First Out (LIFO), i.e. the last ServerActivity implementation that was registered during startup should be the first one to get suspended.
+
+=== Non Requirements
+
+* As discussed previously, in a cloud environment, when the hosting entity crashes, its state should be recovered by the cloud environment without the intervention of the administrator
+* Further developments should be undertaken to modify the graceful shutdown of the EJB subsystem to make it transaction-aware
+
+== Implementation Plan
+
+*Aim of the proposal*. The Transactions subsystem's suspension needs to be modified to to properly delay WildFly’s graceful shutdown as long as there are transactions to complete. Of course, the administrator will be notified when subsystems are delaying WildFly’s suspension. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to the administrator in an understandable manner without exposing internal technical implementation details.
+
+=== Narayana
+
+Narayana will internally handle the lifecycle of transactions during suspension. From the point of view of the integrating party, Narayana should provide a blocking API to suspend itself and it should return control only when there are no transactions left to complete. Moreover, Narayana needs to provide a switch to suspend the creation of new transactions, which needs to be used only when no new transactions are needed.
```

Comment:
You said setting the suspend timeout to a positive value is a way to avoid hanging shutdown if there are heuristics. But the suspend timeout affects every subsystem so I still think we need narayana specific property, then the customer control what the TM is doing without impacting every subsystem. Depending upon how you implement this proposal such a property could be added to narayana so as not to impact the WildFly model.